packages/permissions/src/utils/has-permissions.ts

```diff
@@ -241,25 +247,44 @@ const getHasDemandedDataFence = (options: {
 };
 
 const getDataFenceByTypeAndGroup = (
-  dataFences: TDataFences | null,
+  actualDataFences: TDataFences | null,
   demandedDataFenceType: TDataFenceType,
   demandedDataFenceGroup: string
 ) => {
-  if (!dataFences) return null;
+  if (!actualDataFences) return null;
 
-  if (demandedDataFenceType in dataFences) {
-    const dataFence = dataFences[demandedDataFenceType];
-    if (dataFence && demandedDataFenceGroup in dataFence) {
-      return dataFence[demandedDataFenceGroup];
+  if (demandedDataFenceType in actualDataFences) {
+    const actualDataFence = actualDataFences[demandedDataFenceType];
+    if (actualDataFence && demandedDataFenceGroup in actualDataFence) {
+      return actualDataFence[demandedDataFenceGroup];
     }
   }
   return null;
 };
 
 const getIsPermissionOverwritingDataFence = (
   actualPermissions: TPermissions | null,
-  actualDataFence: TActualDataFence
+  demandedDataFence: TDemandedDataFence
 ) => {
+  if (!actualPermissions) return false;
+
+  /**
+   * NOTE:
+   *    A data fence relates, exists in relation to a resource access.
+   *    This relation is constructed by the group of the data fence.
+   *
+   *    Given the user already has manage access on the data fence's group
```

Comment:
altho I understand you, we are changing the language.

I would stick to using `general permission` and `datafence permission` since all our docs are using these terms.

```
Given user has general permission `manage` on the data fence's group
```

Reply:
do you want to address this first before merging?